File: src/main/java/hilfling/backend/hilfling/controllers/PlaceController.java
package hilfling.backend.hilfling.controllers;

import hilfling.backend.hilfling.model.Place;
import hilfling.backend.hilfling.service.PlaceService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("/api/v1/places")
public class PlaceController extends GenericBaseControllerImplementation<Place> {
    @Autowired
    private PlaceService service;
    public PlaceService getService() {
        return service;
    }
}

File: src/main/kotlin/hilfling/backend/hilfling/controller/PlaceController.kt
package hilfling.backend.hilfling.controller

import hilfling.backend.hilfling.model.Place
import hilfling.backend.hilfling.repository.PlaceRepository
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/places")
class PlaceController {
    val repository = PlaceRepository()

    @GetMapping("/{id}")
    fun getById(@PathVariable("id") id: Int) : Place? {
        return repository.findById(id)
    }

    @GetMapping
    fun getAll() : List<Place> {
        return repository.findAll()
    }

    @PostMapping
    fun create(
            @RequestParam("name") name: String
    ): Place {
        return repository.create(name)
    }
}